support/oidc-discovery-provider/server_api.go

```diff
@@ -79,6 +80,10 @@ func (s *ServerAPISource) FetchKeySet() (*jose.JSONWebKeySet, time.Time, bool) {
 	return s.jwks, s.modTime, true
 }
 
+func (s *ServerAPISource) LastPoll() time.Time {
+	return s.pollTime
+}
```

Comment:
The pollTime can be updated concurrently. LastPoll should take the read-lock of `s.mu` to obtain this value:
```suggestion
func (s *WorkloadAPISource) LastPoll() time.Time {
	s.mu.RLock()
	defer s.mu.RUnlock()
	return s.pollTime
}
```

Reply:
Resolved as suggested